ArduPlane/pullup.cpp

```diff
@@ -0,0 +1,244 @@
+#include "Plane.h"
+/*
+  support for pullup after an alitude wait. Used for high altitude gliders
+ */
+
+#if AP_PLANE_GLIDER_PULLUP_ENABLED
+
+// Pullup control parameters
+const AP_Param::GroupInfo GliderPullup::var_info[] = {
+
+    // @Param: ENABLE
+    // @DisplayName: Enable pullup after altitude wait
+    // @Description: Enable pullup after altitude wait
+    // @Values: 0:Disabled, 1:Enabled
+    // @User: Advanced
+    AP_GROUPINFO_FLAGS("ENABLE", 1, GliderPullup,  enable, 0, AP_PARAM_FLAG_ENABLE),
+
+    // @Param: ELEV_OFS
+    // @DisplayName: Elevator deflection used before starting pullup
+    // @Description: Elevator deflection offset from -1 to 1 while waiting for airspeed to rise before starting close loop control of the pullup.
+    // @Range: -1.0 1.0
+    // @User: Advanced
+    AP_GROUPINFO("ELEV_OFS", 2, GliderPullup,  elev_offset, 0.1f),
+
+    // @Param: NG_LIM
+    // @DisplayName: Maximum normal load factor during pullup
+    // @Description: This is the nominal maximum value of normal load factor used during the closed loop pitch rate control of the pullup.
+    // @Range: 1.0 4.0
+    // @User: Advanced
+    AP_GROUPINFO("NG_LIM", 3, GliderPullup,  ng_limit, 2.0f),
+
+    // @Param: NG_JERK_LIM
+    // @DisplayName: Maximum normal load factor rate of change during pullup
+    // @Description: The normal load factor used for closed loop pitch rate control of the pullup will be ramped up to the value set by PUP_NG_LIM at the rate of change set by this parameter. The parameter value specified will be scaled internally by 1/EAS2TAS.
+    // @Units: 1/s
+    // @Range: 0.1 10.0
+    // @User: Advanced
+    AP_GROUPINFO("NG_JERK_LIM", 4, GliderPullup,  ng_jerk_limit, 4.0f),
+
+    // @Param: PITCH
+    // @DisplayName: Target pitch angle during pullup
+    // @Description: The vehicle will attempt achieve this pitch angle during the pull-up maneouvre.
+    // @Units: deg
+    // @Range: -5 15
+    // @User: Advanced
+    AP_GROUPINFO("PITCH", 5, GliderPullup,  pitch_dem, 3),
+
+    // @Param: ARSPD_START
+    // @DisplayName: Pullup target airspeed
+    // @Description: Target airspeed for initial airspeed wait
+    // @Units: m/s
+    // @Range: 0 100
+    // @User: Advanced
+    AP_GROUPINFO("ARSPD_START", 6, GliderPullup,  airspeed_start, 30),
+
+    // @Param: PITCH_START
+    // @DisplayName: Pullup target pitch
+    // @Description: Target pitch for initial pullup
+    // @Units: deg
+    // @Range: -90 0
```

Comment:
-90 is probably a bad idea, looks like its using eulers.

Reply:
changed to -80